component-generator/utils.js

```diff
@@ -16,6 +18,25 @@ function validateComponentName(value) {
   return value;
 }
 
+/**
+ * Sends request to the Netify function to inform about generate-component usage.
+ * @param {string} componentName - component name
+ */
+function sendTrackInfo(componentName) {
+  const { BASE_URL, TRACK_ANONYMOUS_ANALYTICS } = process.env;
+  if (TRACK_ANONYMOUS_ANALYTICS) {
+    const url = `${BASE_URL}/.netlify/functions/track-generate-component`;
```

Comment:
This should handle the calling of a Netlify function API endpoint, but we will also need to actually create that function/endpoint as well.

Following [the docs here](https://docs.netlify.com/functions/build-with-javascript/) and [this playground](https://functions.netlify.com/playground/), we should be able to add a file for this `netlify/functions` inside of the `www` module, e.g. `www/netlify/functions/trackGenerateComponent.js`. 
This file would be responsible for calling the Segment API to actually dispatch a new event. Something along the lines of:

```js
// www/netlify/functions/trackGenerateComponent.js

exports.handler = async function (event, context) {
  // Only allow POST
  if (event.httpMethod !== "POST") {
    return { statusCode: 405, body: "Method Not Allowed" };
  }
  const { componentName } = event.body; // may need to parse `event.body` first
  // dispatch event to Segment
  analytics.track({
    anonymousId: uuidv4(),
    event: 'edx.paragon.generate-component', // example event name
    properties: { componentName },
  });

  return {
    statusCode: 200,
    body: JSON.stringify({ succes: true }),
  };
}
```

There is [some information on the tools available](https://docs.netlify.com/functions/build-with-javascript/#tools) to help run these functions locally as well, e.g. `netlify dev`.